src/Voron/PageLocator.cs

```diff
@@ -77,9 +49,13 @@ public bool TryGetWritablePage(long pageNumber, out Page page)
 
             ref var node = ref Unsafe.Add(ref _cache[0], (int)bucket);
             page = node.Page;
-            return node.IsWritable && node.PageNumber == pageNumber && node.PageNumber != Invalid;
+            return node.Generation == _generation && node.IsWritable && node.PageNumber == pageNumber && node.PageNumber != Invalid;
         }
 
+        public void Renew()
+        {
+            _generation++;
```

Comment:
What if we use `ushort` for `Generation` here, and then reset the whole array whenever we reach `MaxValue`?
That would bring the `PageData` back to 20 bytes, no?

Reply:
Actually, I think we can use that knowledge and pay big reset only once in 128 attempts (we just fill the memory with 0s and reset the generation to 1) to keep the size as it was. I will see to that tomorrow.

Reply:
Done